server.go

```diff
@@ -383,6 +384,8 @@ func (s *Server) attachClient(cl *Client, listener string) error {
 	}
 
 	s.hooks.OnSessionEstablished(cl, pk)
+	cl.openShutdownSignal()
+	defer cl.sendShutdownSignal()
```

Comment:
Move this to the 1st line in attachClient.

Reply:
My idea here is that if cl.Read() is not executed, there is no need to initiate the signal.

Reply:
Make sense.